### PR TITLE
BJS2-9493 | Pretty error

### PR DESCRIPTION
**Pull request by DROFEL**

### CODE REVIEW

File: src/main/java/faang/school/godbless/error_prettifier/PrettyError.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.function.Supplier;
+
+public class PrettyError{
+    public static <T> T withErrorHandling(Supplier<T> supplier, ExceptionHandler exceptionHandler) {
+        try {
+            return supplier.get();
+        } catch (RuntimeException e) {
+            exceptionHandler.handle(e);
+        }
+//        idk how it can be handled in parametrized function without null
```
Comment (pablo023) [inline src/main/java/faang/school/godbless/error_prettifier/PrettyError.java:12]: how about returning 10th line?

**Review by pablo023 — CHANGES_REQUESTED** [review]

File: src/main/java/faang/school/godbless/error_prettifier/PrettyError.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.function.Supplier;
+
+public class PrettyError{
+    public static <T> T withErrorHandling(Supplier<T> supplier, ExceptionHandler exceptionHandler) {
+        try {
+            return supplier.get();
+        } catch (RuntimeException e) {
+            exceptionHandler.handle(e);
+            throw e;
```
Comment (DROFEL) [inline src/main/java/faang/school/godbless/error_prettifier/PrettyError.java:11]: like this?

Comment (pablo023) [reply src/main/java/faang/school/godbless/error_prettifier/PrettyError.java:11]: Not really. You don't have to throw anything, you can just return exceptionHandler.handle(e)

File: src/main/java/faang/school/godbless/error_prettifier/PrettyError.java (hunk 1 above)
Comment (pablo023) [reply src/main/java/faang/school/godbless/error_prettifier/PrettyError.java]: Aight you could just change the method in the interface to T from void and just return 10th line

**Pull request merged**